src/Query/TermLevel/TermsSetQuery.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchDSL\Query\TermLevel;
+
+use ONGR\ElasticsearchDSL\BuilderInterface;
+use ONGR\ElasticsearchDSL\ParametersTrait;
+
+/**
+ * Represents Elasticsearch "terms_set" query.
+ *
+ * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-set-query.html
+ */
+class TermsSetQuery implements BuilderInterface
+{
+    use ParametersTrait;
+
+    /**
+     * @var string
+     */
+    private $field;
+
+    /**
+     * @var array
+     */
+    private $terms;
+
+    /**
+     * Constructor.
+     *
+     * @param string $field      Field name
+     * @param array  $terms      An array of terms
+     * @param array  $parameters Optional parameters
+     */
+    public function __construct($field, $terms, array $parameters = [])
```

Comment:
This query does not have any point without `minimum_should_match_field`. Please make it a separate required parameter in the constructor

Reply:
@einorler it can `minimum_should_match_field ` or `minimum_should_match_script` 🤔

Reply:
@einorler what is your thoughts on making `$parameters` a required parameter in the constructor and checking for `minimum_should_match_field` or `minimum_should_match_script` existence before calling setParameters.